src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,100 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int CAPACITY = 16;
+    private Node<K, V>[] nodes;
+    private int size;
+
+    public MyHashMap() {
+        nodes = new Node[CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        thresholdCheck();
+        Node<K, V> previous = nodes[getHash(key)];
+        if (previous == null) {
+            nodes[getHash(key)] = new Node<K, V>(key, value, null);
+            size++;
+            return;
+        }
+        while (previous.next != null || Objects.equals(previous.key,key)) {
+            if (Objects.equals(previous.key, key)) {
+                previous.value = value;
+                return;
+            }
+            previous = previous.next;
+        }
+        previous.next = new Node<>(key, value, null);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> node = getNode(key);
+        return (node == null) ? null : node.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node<K, V> getNode(K key) {
+        Node<K, V> node = nodes[getHash(key)];
+        while (node != null) {
+            if (Objects.equals(node.key,key)) {
+                break;
+            }
+            node = node.next;
+        }
+        return node;
+    }
+
+    private void resize() {
+        int newSize = nodes.length * 2;
+        Node<K, V>[] old = new Node[newSize];
```

Comment:
```suggestion
        Node<K, V>[] new = new Node[newSize];
```
why old, you've only created this arr